### PR TITLE
4 Implement Game Class

### LINKED ISSUE #4: Implement the Game Class
The Game class will handle the result of a blackjack game from start to finish, calling the dealer and player class functions for their decisions.

The main game function will take in the betting amount as an integer, and will return a dictionary including the result of the bet (how much was won/lost per game), and more data as per the following sample:

```
game_info = {
    "result": 0,
    "player_sum": [0],
    "dealer_sum": 0,
    "blackjack": 0,
    "double_down": 0,
    "split": 0
}
```

Where the player_sum is a list of sums for each hand the player has in the round (to account for splits)

blackjack. double_down, and split are the number of occurrences of each of those events per round.

### PR DESCRIPTION
**Pull request by esulu**
Created the Game class which handles the result of a blackjack game from start to finish, calling the dealer and player class functions for their decisions.

Additionally created a test script, which outputs various game decisions/outcomes into a text file. This tests every possible player starting hand, with randomized dealer hands drawn for each case. 

Resolves #4 

**Reviewers**
- [x] @cailynse 
- [x] @nwarsalee

### CODE REVIEW

**Comment by esulu** [comment]
Todo: deal 2 cards to the dealer initially

**Comment by esulu** [comment]
Todo: rework split scenario to compute dealer decision only after both hands bust/stand

File: game.py [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+
+                # player busts on this draw and loses betting amount
+                if player_sum > 21:
+                    return self.make_result_dict(
+                        result      = -1 * bet, 
+                        player_sum  = [player_sum],
+                        double_down = double_down)
+
+            # split
+            if player_action == definitions.Actions.SPLIT:
```
Comment (esulu) [inline game.py:139]: TODO: Rework the splitting to play dealer's part after all player decisions

File: game.py [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+                
+                # play out the non-completed player hands
+                for i in range(2):
+                    if type(results[i]) is list:
+                        results[i] = self.compute_final_result(results[i][0], dealer_hand, deck, results[i][1], results[i][2])
+
+                # combine the two into a single dict
+                return self.make_result_dict(
+                    result      = results[0]["result"] + results[1]["result"], 
+                    player_sum  = results[0]["player_sum"] + results[1]["player_sum"],
```
Comment (nwarsalee) [inline game.py:154]: Wouldn't the Player's sum here possibly end up totaling over 21 here if both hands sums are being added?

**Review by nwarsalee — APPROVED** [review]
Looks good 👍

File: game.py (hunk 2 above)
Comment (esulu) [reply game.py:154]: player_sum is a list, the + operation concatenates the two lists

File: README.md [hunk 3]
```diff
@@ -32,7 +32,7 @@ From these multiple trials, we extract the win/lose probabilities for both Deale
 # Blackjack Rules Assumed
 - Single deck
 - Dealer stands on any 17 (including soft 17)
-- No hole card: dealer does not consult second card until after the player's final decision
+- No hole card: dealer does draw nor consult their second card until after the player's final decision
```
Comment (nwarsalee) [inline README.md]: dealer does or does not?

**Review by cailyn-codes — APPROVED** [review]

**Pull request merged**